src/ast/nodes/IfStatement.ts

```diff
@@ -45,33 +50,43 @@ export default class IfStatement extends StatementBase {
 	}
 
 	render(code: MagicString, options: RenderOptions) {
-		const testValue = this.test.getValue();
+		// Note that unknown test values are always included
+		const testValue = this.hasUnknownTestValue ? UNKNOWN_VALUE : this.test.getValue();
 		if (
-			!this.context.treeshake ||
-			this.test.included ||
-			(testValue ? this.alternate !== null && this.alternate.included : this.consequent.included)
+			!this.test.included &&
+			(testValue ? this.alternate === null || !this.alternate.included : !this.consequent.included)
 		) {
-			super.render(code, options);
+			const singleRetainedBranch = testValue ? this.consequent : this.alternate;
+			code.remove(this.start, singleRetainedBranch.start);
+			code.remove(singleRetainedBranch.end, this.end);
+			singleRetainedBranch.render(code, options);
 		} else {
-			// if test is not included, it is impossible that alternate===null even though it is the retained branch
-			const branchToRetain = testValue ? this.consequent : this.alternate;
-			code.remove(this.start, branchToRetain.start);
-			code.remove(branchToRetain.end, this.end);
-			branchToRetain.render(code, options);
+			if (this.test.included) {
+				this.test.render(code, options);
+			} else {
+				code.overwrite(this.test.start, this.test.end, JSON.stringify(testValue));
```

Comment:
Is this our first implementation of value inlining? 😮

Although we probably wouldn't want this if values ever track more complex objects. That said `getValueAtPath` neatly avoids this problem.

Reply:
What if the value was a really long string? Would we still be inlining it?

Also surely we only need to inline the truthiness or falsyness of the value over the actual value itself?

Reply:
Yes, `getValue(AtPath)` can and only ever should return primitive values or `UNKNOWN_VALUE`. That being said, you are right, it would make much more sense to just always write `true` or `false` here.